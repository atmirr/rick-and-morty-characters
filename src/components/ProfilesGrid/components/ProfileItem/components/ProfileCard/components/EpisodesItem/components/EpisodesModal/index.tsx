import React from 'react';
import styled from 'styled-components';
import Modal from './components/Modal';

const ModalTitle = styled.h4`
  padding: 5px 0;
  font-size: 20px;
  font-weight: 600;
`;

const ModalWrapper = styled.div`
  background-color: #ffffff;
  text-align: center;
  width: 400px;
  max-width: 100%;
  height: 500px;
  max-height: 100%;
  position: fixed;
  z-index: 999;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 1.5em 1.3em;
  margin: 1.1em 0.9em;
  border-radius: 15px;
  overflow: auto;
  cursor: default;
`;

const EpisodesList = styled.ul`
  text-align: left;
  list-style-type: none;
`;

const EpisodeItem = styled.li`
  padding: 5px 0;
`;

type PropTypes = {
  episodes?: string[];
  handleCloseModal: () => void;
};

function EpisodesModal({
  episodes = [],
  handleCloseModal,
}: PropTypes): React.ReactElement {
  return (
    <Modal defaultOpened fade handleCloseModal={handleCloseModal}>
      <ModalWrapper>
        <ModalTitle>List of Featured Episodes</ModalTitle>
        <EpisodesList>
          {episodes.map((episode, index) => (
            <EpisodeItem key={index}>{episode}</EpisodeItem>
          ))}
        </EpisodesList>
      </ModalWrapper>
    </Modal>
  );
}

export default EpisodesModal;
