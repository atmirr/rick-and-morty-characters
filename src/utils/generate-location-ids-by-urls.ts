export const generateLocationIdsByUrls = (urls: string[]): string[] =>
  urls.reduce((result: string[], url: string) => {
    const splittedUrl = url.split('/');
    const id = splittedUrl[splittedUrl.length - 1];
    return id ? [...result, id] : [...result];
  }, []);
