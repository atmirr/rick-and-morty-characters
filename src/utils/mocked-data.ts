export const characters = [
  {
    id: 1,
    name: 'Rick Sanchez',
    species: 'Human',
    gender: 'Male',
    created: '2017-11-04T18:48:46.250Z',
    episode: [
      'https://rickandmortyapi.com/api/episode/1',
      'https://rickandmortyapi.com/api/episode/2',
    ],
    image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
    location: {
      name: 'Earth (Replacement Dimension)',
      url: 'https://rickandmortyapi.com/api/location/20',
    },
    origin: {
      name: 'Earth (C-137)',
      url: 'https://rickandmortyapi.com/api/location/1',
    },
    status: 'Alive',
    type: '',
    url: 'https://rickandmortyapi.com/api/character/1',
  },
  {
    id: 2,
    name: 'Morty Smith',
    species: 'Human',
    gender: 'Male',
    created: '2017-11-04T18:50:21.651Z',
    episode: [
      'https://rickandmortyapi.com/api/episode/1',
      'https://rickandmortyapi.com/api/episode/2',
      'https://rickandmortyapi.com/api/episode/3',
    ],
    image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
    location: {
      name: 'Earth (Replacement Dimension)',
      url: 'https://rickandmortyapi.com/api/location/20',
    },
    origin: {
      name: 'Earth (C-137)',
      url: 'https://rickandmortyapi.com/api/location/1',
    },
    status: 'Alive',
    type: '',
    url: 'https://rickandmortyapi.com/api/character/2',
  },
];

export const episodes = [
  'Rixty Minutes',
  'Something Ricked This Way Comes',
  'Close Rick-counters of the Rick Kind',
  'Ricksy Business',
  'A Rickle in Time',
];

export const locations = [
  {
    name: 'Citadel of Ricks',
    type: 'Space station',
    dimension: 'unknown',
  },
  {
    name: 'Earth (Replacement Dimension)',
    type: 'Planet',
    dimension: 'Replacement Dimension',
  },
];
