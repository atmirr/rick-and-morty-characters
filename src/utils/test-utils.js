import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockedInitialState = {
  characters: { items: [], nextPage: null, error: null, isLoading: false },
  episodes: { items: [], nextPage: null, error: null, isLoading: false },
  locations: { items: [], error: null, isLoading: false },
};

function render(
  ui,
  {
    initialState = mockedInitialState,
    store = configureStore()(initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
