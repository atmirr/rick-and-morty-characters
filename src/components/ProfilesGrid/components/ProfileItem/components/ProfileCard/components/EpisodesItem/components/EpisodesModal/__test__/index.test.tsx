import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { screen, render } from 'src/utils/test-utils';
import { episodes } from 'src/utils/mocked-data';
import EpisodesModal from '../index';

describe('<EpisodesModal />', () => {
  it('should renders the component', () => {
    const { container } = render(<EpisodesModal />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display the modal by default', () => {
    render(<EpisodesModal />);
    expect(screen.getAllByTestId('modal').length).toBe(1);
    expect(screen.queryAllByText(/List of featured episodes/i).length).toBe(1);
  });

  it('should display the list of the episodes if the episodes props has been set', () => {
    render(<EpisodesModal episodes={episodes} />);
    expect(screen.getAllByTestId('modal').length).toBe(1);
    expect(screen.queryAllByText(/List of featured episodes/i).length).toBe(1);
    episodes.forEach((episode) => {
      expect(screen.queryAllByText(episode).length).toBe(1);
    });
  });
});
