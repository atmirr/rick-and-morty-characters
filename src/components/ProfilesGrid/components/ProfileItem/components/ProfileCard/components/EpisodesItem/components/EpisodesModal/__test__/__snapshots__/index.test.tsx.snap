// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodesModal /> should renders the component 1`] = `null`;
